xprof/btx_timeline.cpp

```diff
@@ -153,31 +203,59 @@ static void add_event_DTelemetry(timeline_dispatch_t *dispatch, std::string host
   track_event->set_double_counter_value(value);
 }
 
+static void add_event_memModule( timeline_dispatch_t *dispatch, std::string hostname,
+                                 uint64_t process_id, uint64_t thread_id, uint64_t did, uint32_t deviceIdx, uintptr_t hMemModule, uint32_t subDevice, uint64_t timestamp, float pBandwidth, float rdBandwidth, float wtBandwidth, float occupancy) {
+  // Define details for RX throughput.
+  Details details = {false, 0, 0};
+  add_event_DTelemetry(dispatch, hostname, process_id, thread_id, did, deviceIdx, hMemModule, subDevice, timestamp,
+                       rdBandwidth, get_Bandwidth_track_uuuid, "Memory Read BW", details);
+  details.RxTx = true;
+  add_event_DTelemetry(dispatch, hostname, process_id, thread_id, did, deviceIdx, hMemModule, subDevice, timestamp,
+                       wtBandwidth, get_Bandwidth_track_uuuid, "Memory Write BW", details);
+  add_event_DTelemetry(dispatch, hostname, process_id, thread_id, did, deviceIdx, hMemModule, subDevice, timestamp,
+                       occupancy, get_Occupancy_track_uuuid, "Memory Occupancy", details);
+}
+
+
+static void add_event_fabricPort( timeline_dispatch_t *dispatch, std::string hostname,
+                                 uint64_t process_id, uint64_t thread_id, uint64_t did, uint32_t deviceIdx, uintptr_t hFabricPort,
+                                 uint32_t subDevice, uint64_t timestamp, uint32_t fabricId, uint32_t remotePortId, 
+                                 float rxThroughput, float txThroughput, float rxSpeed, float txSpeed) {
+  // Define details for RX throughput.
+  Details details = {false, fabricId, remotePortId};
+  add_event_DTelemetry(dispatch, hostname, process_id, thread_id, did, deviceIdx, hFabricPort, subDevice, timestamp,
+                       rxThroughput, get_fpThroughput_track_uuuid, "Fabric ThroughputRX", details);
+
+  details.RxTx = true;
+  add_event_DTelemetry(dispatch, hostname, process_id, thread_id, did, deviceIdx, hFabricPort, subDevice, timestamp,
```

Comment:
Please add some documentation somewhere to explain `details`. What information he has and how it's useful it it's.